### PR TITLE
replace references of https://storage.googleapis.com/kubernetes-release with https://dl.k8s.io

### PR DESCRIPTION
**Pull request by KlwntSingh**
fixes: https://github.com/kubernetes/kops/issues/15402

### CODE REVIEW

File: upup/pkg/fi/cloudup/defaults.go [hunk 1]
```diff
@@ -167,10 +167,10 @@ func ensureKubernetesVersion(c *kops.Cluster) error {
 }
 
 // FindLatestKubernetesVersion returns the latest kubernetes version,
-// as stored at https://storage.googleapis.com/kubernetes-release/release/stable.txt
+// as stored at https://dl.k8s.io/release/stable.txt
 // This shouldn't be used any more; we prefer reading the stable channel
 func FindLatestKubernetesVersion() (string, error) {
-	stableURL := "https://storage.googleapis.com/kubernetes-release/release/stable.txt"
+	stableURL := "https://dl.k8s.io/release/stable.txt"
```
Comment (hakman) [inline upup/pkg/fi/cloudup/defaults.go:173]: Not quite sure what's wrong here:
```
% curl https://storage.googleapis.com/kubernetes-release/release/stable.txt
v1.27.1

% curl https://dl.k8s.io/release/stable.txt                                
<html>
<head><title>302 Found</title></head>
<body>
<center><h1>302 Found</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

Comment (rjsadow) [reply upup/pkg/fi/cloudup/defaults.go:173]: Try adding a `-L` to your curl

Comment (rjsadow) [reply upup/pkg/fi/cloudup/defaults.go:173]: We should ensure redirects are used anywhere `stableURL` is referenced.

**Comment by rifelpet** [comment]
It looks like you need to run ./hack/update-expected.sh to pick up more integration test changes.

We'll also want to update our e2e tests to reference the new location too: https://github.com/kubernetes/test-infra/blob/2326b828d25f04c676703ceda0782a3eed596a12/config/jobs/kubernetes/kops/helpers.py#L78

File: upup/pkg/fi/cloudup/defaults.go (hunk 1 above)
Comment (hakman) [reply upup/pkg/fi/cloudup/defaults.go:173]: Thanks @rjsadow. 😁

**Review by hakman — APPROVED** [review]

**Pull request merged**